### ISSUE TITLE
Air Conditioner Support

### ISSUE THREAD

**Issue by nitaybz** [KhaosT/HAP-NodeJS#461]
are you guys going to add the characteristic AirConditioner?
https://developer.apple.com/documentation/homekit/hmaccessorycategorytypeairconditioner

**Comment by jetmcquack** [comment]
+1 ! would be greatly appreciated!

**Comment by KhaosT** [comment]
This is being listed as HeaterCooler service. See [here](https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L2792).

**Comment by nitaybz** [comment]
@KhaosT Thanks for your answer, I did check up on that service and there are couple of things that seems weird to me.
first of all I'm missing some characteristics to make this service useful, like:
TargetTemperature
CurrentRelativeHumidity

and I would be happy to use those, which appear in the link I shared above:
CurrentFanState
TargetFanState
CurrentHumidifierDehumidifierState
TargetHumidifierDehumidifierState

I only saw them in another services, can't they be added to the same service?

and one last thing, it appears that in Home app it shows "Not Supported" but I can definitely control the device with Siri or with other Homekit apps.
is it possible that it still not supported in home app even though there are already 2 AC products in the market that supports Homekit??

**Comment by KhaosT** [comment]
@nitaybz I believe all those AC products are using thermostat profile since technically they do the same thing?

**Comment by nitaybz** [comment]
the reason I was looking for using the AirConditioner characteristic was because I wanted to use the Swing and rotation functions. I believe those ACs that are connected to homekit are using the Air~Conditioner characteristic according to some screenshots I've seen.
is it possible to add Swing or RotationSpeed to thermostat?
or can you add the others I requested to the HeaterCooler Service?

**Comment by nitaybz** [comment]
@KhaosT should I open a new issue? regarding adding the extra characteristics to HeaterCoolerService...

+ I just found out (after updating to iOS 11) that this service is showing correctly, so I really want to work on making this service useful

**Comment by KhaosT** [comment]
You can always add any characteristics to a service. The problem is HomeKit
app may not show UI properly that allows you to interact with your added
characteristics.

And what's your issue on adding characteristics to a service?

**Comment by nitaybz** [comment]
this is the error I get when I try to set a target temperature on that service:
"TypeError: Cannot read property 'setProps' of undefined"

this is how I used the service:
```
    this.service = new Service.HeaterCooler(this.name);
        
    this.service.getCharacteristic(Characteristic.CurrentHeaterCoolerState)
        .on('get', this.getFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.TargetHeaterCoolerState)
        .on('get', this.getFalseCallback.bind(this))
        .on('set', this.setFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.CurrentTemperature)
        .setProps({
            minValue: 0,
            maxValue: 100,
            minStep: 0.01
        })
        .on('get', this.getFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.TargetTemperature)
        .setProps({
            minValue: minValue,
            maxValue: maxValue,
            minStep: 1
        })
        .on('get', this.getFalseCallback.bind(this))
        .on('set', this.setFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.TemperatureDisplayUnits)
        .on('get', this.getFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.CurrentRelativeHumidity)
        .setProps({
            minValue: 0,
            maxValue: 100,
            minStep: 0.01
        })
        .on('get', this.getFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.SwingMode)
        .on('get', this.getFalseCallback.bind(this))
        .on('set', this.setFalseCallback.bind(this));

    this.service.getCharacteristic(Characteristic.RotationSpeed)
        .setProps({
                minValue: 0,
                maxValue: 100,
                minStep: 1
            })
        .on('get', this.getFalseCallback.bind(this))
        .on('set', this.setFalseCallback.bind(this));
        
    this.service.addCharacteristic(Characteristic.Active)
        .on('get', this.getFalseCallback.bind(this))
        .on('set', this.setFalseCallback.bind(this));

    return [informationService, this.service];
}
```

this error comes up on the Target temperature characteristic, which works perfectly this way in the thermostat.
same error appears for the CurrentRelativeHumidity.

**Comment by peros550** [comment]
Hey again, while I was troubleshooting I decided to flash Maxim's thermostat code (this was my starting point in trying to build an A/C example) and had the same unresponsive situation when closed and reopened the Home app. 

Same problem has been discussed here. 
https://github.com/maximkulkin/esp-homekit-demo/issues/108

 I very confused to say the least. I manually deleted all components in folder "esp-homekit-demo/components" and ran command "git submodule update --init --recursive" to get all submodules updated. 

Based on the discussion on the above issue (108) this issue has been fixed and there were people who reported positive news. That's why I am so confused :(
